script.js

```diff
@@ -1,12 +1,602 @@
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  freeSearchResultShows.innerHTML = freeSearchResultShowsHtml;
+
+}
+
+const changeFormSection = () => {
+  if (freeSearchCont.style.display === 'flex' && seasonInputContainer.style.display === 'none') {
+    freeSearchCont.style.display = 'none';
+    allShowsContainer.style.display = 'none';
+    castListSection.style.display = 'flex';
+    seasonInputContainer.style.display = 'flex';
+  }
+  freeSearchCont.style.display = 'flex';
+  castListSection.style.display = 'none';
+  allShowsContainer.style.display = 'block';
+  seasonInputContainer.style.display = 'none';
+};
+
+const getGenreListing = () => {
+  let genresList = [];
+  let allGenres = [];
+  let showGenresObject = {};
+
+  for (let i = 0; i < allShows.length; i++){
+    let showGenres = allShows[i].genres;
+    for (let j = 0; j < showGenres.length; j++){
+      if (!genresList.includes(showGenres[j])){
+        let currentGenre = showGenres[j];
+        
+        genresList.push(showGenres[j])
+        showGenresObject[`${currentGenre}`] = allShows.filter((show) => {
+          if(show.genres.includes(currentGenre)){
+            return show
+          }
+        });
+      }
+    }
+  }
+  allGenres.push(showGenresObject);
+  return allGenres.sort();
+};
+
+function onLoadScreen(shows) {
+  freeSearchCont.style.display = 'flex';
+  showContainer.style.display = 'none';
+  seriesHeader.style.display = 'none';
+  let pageHeaderTitle = document.getElementById('pageHeaderTitle')
+  pageHeaderTitle.innerText = 'Shows';
+  let topRatedShows = document.getElementById('topRatedShows');
+  let dramaShows = document.getElementById('dramaShows');
+  let comedyShows = document.getElementById('comedyShows');
+  let crimeShows = document.getElementById('crimeShows');
+  let actionShows = document.getElementById('actionShows');
+  let scifiShows = document.getElementById('scifiShows');
+
+  let topRatedShowsHtml = shows.slice([0],[10]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  topRatedShows.innerHTML = topRatedShowsHtml;
+  
+  const allShowGenres = getGenreListing();
+  const allDramaShows = allShowGenres[0].Drama;
```

Comment:
You could use destructuring at this point to get all the genres - e.g.

`let { Drama: allDramaShows, Comedy: allComedyShow, ... , "Science-Fiction": allSciFiShows } = allShowGenres[0];`